{
  "uid" : "482a294cccb57025",
  "name" : "TC 02.02.05 - Check Contact Us link in footer leads to the correct page from each page specified in DATA_1",
  "fullName" : "tests.test_footer.TestFooter#test_tc_02_02_05_check_contact_us_link_functionality",
  "historyId" : "94f0e1cf98dd00a5517d7dafd8a82281",
  "time" : {
    "start" : 1690073992847,
    "stop" : 1690073998982,
    "duration" : 6135
  },
  "description" : "Check that Contact Us link in footer is correct",
  "descriptionHtml" : "<p>Check that Contact Us link in footer is correct</p>\n",
  "status" : "broken",
  "statusMessage" : "selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x56213bdcb4e3 <unknown>\n#1 0x56213bafac76 <unknown>\n#2 0x56213bb36c96 <unknown>\n#3 0x56213bb36dc1 <unknown>\n#4 0x56213bb707f4 <unknown>\n#5 0x56213bb5603d <unknown>\n#6 0x56213bb6e30e <unknown>\n#7 0x56213bb55de3 <unknown>\n#8 0x56213bb2b2dd <unknown>\n#9 0x56213bb2c34e <unknown>\n#10 0x56213bd8b3e4 <unknown>\n#11 0x56213bd8f3d7 <unknown>\n#12 0x56213bd99b20 <unknown>\n#13 0x56213bd90023 <unknown>\n#14 0x56213bd5e1aa <unknown>\n#15 0x56213bdb46b8 <unknown>\n#16 0x56213bdb4847 <unknown>\n#17 0x56213bdc4243 <unknown>\n#18 0x7f992e3eeb43 <unknown>",
  "statusTrace" : "self = <test_footer.TestFooter object at 0x7fcb8c871210>\ndriver = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"b90739a15c7e07786a26676fbd44ca9d\")>\nURL = 'https://magento.softwaretestingboard.com/what-is-new.html'\n\n    @allure.title(\"TC 02.02.05 - Check Contact Us link in footer leads to the correct page \"\n                  \"from each page specified in DATA_1\")\n    @pytest.mark.parametrize('URL', DATA_1)\n    def test_tc_02_02_05_check_contact_us_link_functionality(self, driver, URL):\n        \"\"\"Check that Contact Us link in footer is correct\"\"\"\n        page = FooterPage(driver, url=URL)\n        page.open()\n>       page.check_contact_us_link_functionality()\n\ntests/test_footer.py:501: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npages/footer_page.py:305: in check_contact_us_link_functionality\n    contact_us_link = self.element_is_visible(self.footer_locators.CONTACT_US_LINK)\npages/base_page.py:29: in element_is_visible\n    self.go_to_element(self.element_is_present(locator))\npages/base_page.py:50: in element_is_present\n    return wait(self.driver, timeout).until(EC.presence_of_element_located(locator))\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"b90739a15c7e07786a26676fbd44ca9d\")>\nmethod = <function presence_of_element_located.<locals>._predicate at 0x7fcb8d75a980>\nmessage = ''\n\n    def until(self, method, message: str = \"\"):\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            time.sleep(self._poll)\n            if time.monotonic() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: \nE       Stacktrace:\nE       #0 0x56213bdcb4e3 <unknown>\nE       #1 0x56213bafac76 <unknown>\nE       #2 0x56213bb36c96 <unknown>\nE       #3 0x56213bb36dc1 <unknown>\nE       #4 0x56213bb707f4 <unknown>\nE       #5 0x56213bb5603d <unknown>\nE       #6 0x56213bb6e30e <unknown>\nE       #7 0x56213bb55de3 <unknown>\nE       #8 0x56213bb2b2dd <unknown>\nE       #9 0x56213bb2c34e <unknown>\nE       #10 0x56213bd8b3e4 <unknown>\nE       #11 0x56213bd8f3d7 <unknown>\nE       #12 0x56213bd99b20 <unknown>\nE       #13 0x56213bd90023 <unknown>\nE       #14 0x56213bd5e1aa <unknown>\nE       #15 0x56213bdb46b8 <unknown>\nE       #16 0x56213bdb4847 <unknown>\nE       #17 0x56213bdc4243 <unknown>\nE       #18 0x7f992e3eeb43 <unknown>\n\n/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/selenium/webdriver/support/wait.py:95: TimeoutException",
  "flaky" : false,
  "newFailed" : false,
  "beforeStages" : [ {
    "name" : "_session_faker",
    "time" : {
      "start" : 1690073506870,
      "stop" : 1690073506962,
      "duration" : 92
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "clear_allure_result_folder",
    "time" : {
      "start" : 1690073506962,
      "stop" : 1690073506962,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "driver",
    "time" : {
      "start" : 1690073991822,
      "stop" : 1690073992846,
      "duration" : 1024
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  } ],
  "testStage" : {
    "description" : "Check that Contact Us link in footer is correct",
    "status" : "broken",
    "statusMessage" : "selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x56213bdcb4e3 <unknown>\n#1 0x56213bafac76 <unknown>\n#2 0x56213bb36c96 <unknown>\n#3 0x56213bb36dc1 <unknown>\n#4 0x56213bb707f4 <unknown>\n#5 0x56213bb5603d <unknown>\n#6 0x56213bb6e30e <unknown>\n#7 0x56213bb55de3 <unknown>\n#8 0x56213bb2b2dd <unknown>\n#9 0x56213bb2c34e <unknown>\n#10 0x56213bd8b3e4 <unknown>\n#11 0x56213bd8f3d7 <unknown>\n#12 0x56213bd99b20 <unknown>\n#13 0x56213bd90023 <unknown>\n#14 0x56213bd5e1aa <unknown>\n#15 0x56213bdb46b8 <unknown>\n#16 0x56213bdb4847 <unknown>\n#17 0x56213bdc4243 <unknown>\n#18 0x7f992e3eeb43 <unknown>",
    "statusTrace" : "self = <test_footer.TestFooter object at 0x7fcb8c871210>\ndriver = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"b90739a15c7e07786a26676fbd44ca9d\")>\nURL = 'https://magento.softwaretestingboard.com/what-is-new.html'\n\n    @allure.title(\"TC 02.02.05 - Check Contact Us link in footer leads to the correct page \"\n                  \"from each page specified in DATA_1\")\n    @pytest.mark.parametrize('URL', DATA_1)\n    def test_tc_02_02_05_check_contact_us_link_functionality(self, driver, URL):\n        \"\"\"Check that Contact Us link in footer is correct\"\"\"\n        page = FooterPage(driver, url=URL)\n        page.open()\n>       page.check_contact_us_link_functionality()\n\ntests/test_footer.py:501: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npages/footer_page.py:305: in check_contact_us_link_functionality\n    contact_us_link = self.element_is_visible(self.footer_locators.CONTACT_US_LINK)\npages/base_page.py:29: in element_is_visible\n    self.go_to_element(self.element_is_present(locator))\npages/base_page.py:50: in element_is_present\n    return wait(self.driver, timeout).until(EC.presence_of_element_located(locator))\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"b90739a15c7e07786a26676fbd44ca9d\")>\nmethod = <function presence_of_element_located.<locals>._predicate at 0x7fcb8d75a980>\nmessage = ''\n\n    def until(self, method, message: str = \"\"):\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            time.sleep(self._poll)\n            if time.monotonic() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: \nE       Stacktrace:\nE       #0 0x56213bdcb4e3 <unknown>\nE       #1 0x56213bafac76 <unknown>\nE       #2 0x56213bb36c96 <unknown>\nE       #3 0x56213bb36dc1 <unknown>\nE       #4 0x56213bb707f4 <unknown>\nE       #5 0x56213bb5603d <unknown>\nE       #6 0x56213bb6e30e <unknown>\nE       #7 0x56213bb55de3 <unknown>\nE       #8 0x56213bb2b2dd <unknown>\nE       #9 0x56213bb2c34e <unknown>\nE       #10 0x56213bd8b3e4 <unknown>\nE       #11 0x56213bd8f3d7 <unknown>\nE       #12 0x56213bd99b20 <unknown>\nE       #13 0x56213bd90023 <unknown>\nE       #14 0x56213bd5e1aa <unknown>\nE       #15 0x56213bdb46b8 <unknown>\nE       #16 0x56213bdb4847 <unknown>\nE       #17 0x56213bdc4243 <unknown>\nE       #18 0x7f992e3eeb43 <unknown>\n\n/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/selenium/webdriver/support/wait.py:95: TimeoutException",
    "steps" : [ {
      "name" : "Open a browser",
      "time" : {
        "start" : 1690073992847,
        "stop" : 1690073993841,
        "duration" : 994
      },
      "status" : "passed",
      "steps" : [ ],
      "attachments" : [ ],
      "parameters" : [ ],
      "stepsCount" : 0,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : false,
      "hasContent" : false
    }, {
      "name" : "Check Contact Us link leads to the correct page",
      "time" : {
        "start" : 1690073993841,
        "stop" : 1690073998981,
        "duration" : 5140
      },
      "status" : "broken",
      "statusMessage" : "selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x56213bdcb4e3 <unknown>\n#1 0x56213bafac76 <unknown>\n#2 0x56213bb36c96 <unknown>\n#3 0x56213bb36dc1 <unknown>\n#4 0x56213bb707f4 <unknown>\n#5 0x56213bb5603d <unknown>\n#6 0x56213bb6e30e <unknown>\n#7 0x56213bb55de3 <unknown>\n#8 0x56213bb2b2dd <unknown>\n#9 0x56213bb2c34e <unknown>\n#10 0x56213bd8b3e4 <unknown>\n#11 0x56213bd8f3d7 <unknown>\n#12 0x56213bd99b20 <unknown>\n#13 0x56213bd90023 <unknown>\n#14 0x56213bd5e1aa <unknown>\n#15 0x56213bdb46b8 <unknown>\n#16 0x56213bdb4847 <unknown>\n#17 0x56213bdc4243 <unknown>\n#18 0x7f992e3eeb43 <unknown>\n\n",
      "statusTrace" : "  File \"/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/luma_project/luma_project/pages/footer_page.py\", line 305, in check_contact_us_link_functionality\n    contact_us_link = self.element_is_visible(self.footer_locators.CONTACT_US_LINK)\n                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/luma_project/luma_project/pages/base_page.py\", line 29, in element_is_visible\n    self.go_to_element(self.element_is_present(locator))\n                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/luma_project/luma_project/pages/base_page.py\", line 50, in element_is_present\n    return wait(self.driver, timeout).until(EC.presence_of_element_located(locator))\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/selenium/webdriver/support/wait.py\", line 95, in until\n    raise TimeoutException(message, screen, stacktrace)\n",
      "steps" : [ {
        "name" : "Find a visible element",
        "time" : {
          "start" : 1690073993841,
          "stop" : 1690073998981,
          "duration" : 5140
        },
        "status" : "broken",
        "statusMessage" : "selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x56213bdcb4e3 <unknown>\n#1 0x56213bafac76 <unknown>\n#2 0x56213bb36c96 <unknown>\n#3 0x56213bb36dc1 <unknown>\n#4 0x56213bb707f4 <unknown>\n#5 0x56213bb5603d <unknown>\n#6 0x56213bb6e30e <unknown>\n#7 0x56213bb55de3 <unknown>\n#8 0x56213bb2b2dd <unknown>\n#9 0x56213bb2c34e <unknown>\n#10 0x56213bd8b3e4 <unknown>\n#11 0x56213bd8f3d7 <unknown>\n#12 0x56213bd99b20 <unknown>\n#13 0x56213bd90023 <unknown>\n#14 0x56213bd5e1aa <unknown>\n#15 0x56213bdb46b8 <unknown>\n#16 0x56213bdb4847 <unknown>\n#17 0x56213bdc4243 <unknown>\n#18 0x7f992e3eeb43 <unknown>\n\n",
        "statusTrace" : "  File \"/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/luma_project/luma_project/pages/base_page.py\", line 29, in element_is_visible\n    self.go_to_element(self.element_is_present(locator))\n                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/luma_project/luma_project/pages/base_page.py\", line 50, in element_is_present\n    return wait(self.driver, timeout).until(EC.presence_of_element_located(locator))\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/selenium/webdriver/support/wait.py\", line 95, in until\n    raise TimeoutException(message, screen, stacktrace)\n",
        "steps" : [ {
          "name" : "Find a present element",
          "time" : {
            "start" : 1690073993841,
            "stop" : 1690073998980,
            "duration" : 5139
          },
          "status" : "broken",
          "statusMessage" : "selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x56213bdcb4e3 <unknown>\n#1 0x56213bafac76 <unknown>\n#2 0x56213bb36c96 <unknown>\n#3 0x56213bb36dc1 <unknown>\n#4 0x56213bb707f4 <unknown>\n#5 0x56213bb5603d <unknown>\n#6 0x56213bb6e30e <unknown>\n#7 0x56213bb55de3 <unknown>\n#8 0x56213bb2b2dd <unknown>\n#9 0x56213bb2c34e <unknown>\n#10 0x56213bd8b3e4 <unknown>\n#11 0x56213bd8f3d7 <unknown>\n#12 0x56213bd99b20 <unknown>\n#13 0x56213bd90023 <unknown>\n#14 0x56213bd5e1aa <unknown>\n#15 0x56213bdb46b8 <unknown>\n#16 0x56213bdb4847 <unknown>\n#17 0x56213bdc4243 <unknown>\n#18 0x7f992e3eeb43 <unknown>\n\n",
          "statusTrace" : "  File \"/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/luma_project/luma_project/pages/base_page.py\", line 50, in element_is_present\n    return wait(self.driver, timeout).until(EC.presence_of_element_located(locator))\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/selenium/webdriver/support/wait.py\", line 95, in until\n    raise TimeoutException(message, screen, stacktrace)\n",
          "steps" : [ ],
          "attachments" : [ ],
          "parameters" : [ {
            "name" : "locator",
            "value" : "('css selector', '.footer > :nth-child(5) > a')"
          }, {
            "name" : "timeout",
            "value" : "5"
          } ],
          "stepsCount" : 0,
          "attachmentsCount" : 0,
          "shouldDisplayMessage" : true,
          "hasContent" : true
        } ],
        "attachments" : [ ],
        "parameters" : [ {
          "name" : "locator",
          "value" : "('css selector', '.footer > :nth-child(5) > a')"
        }, {
          "name" : "timeout",
          "value" : "10"
        } ],
        "stepsCount" : 1,
        "attachmentsCount" : 0,
        "shouldDisplayMessage" : false,
        "hasContent" : true
      } ],
      "attachments" : [ ],
      "parameters" : [ ],
      "stepsCount" : 2,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : false,
      "hasContent" : true
    } ],
    "attachments" : [ {
      "uid" : "7a38f93ee650b7b0",
      "name" : "Screenshot",
      "source" : "7a38f93ee650b7b0.png",
      "type" : "image/png",
      "size" : 799480
    }, {
      "uid" : "cab8d1a169776ce2",
      "name" : "HTML source",
      "source" : "cab8d1a169776ce2.html",
      "type" : "text/html",
      "size" : 113065
    } ],
    "parameters" : [ ],
    "stepsCount" : 4,
    "attachmentsCount" : 2,
    "shouldDisplayMessage" : true,
    "hasContent" : true
  },
  "afterStages" : [ {
    "name" : "clear_allure_result_folder::0",
    "time" : {
      "start" : 1690075074103,
      "stop" : 1690075074103,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "driver::0",
    "time" : {
      "start" : 1690073999330,
      "stop" : 1690073999396,
      "duration" : 66
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  } ],
  "labels" : [ {
    "name" : "epic",
    "value" : "Test Footer"
  }, {
    "name" : "parentSuite",
    "value" : "tests"
  }, {
    "name" : "suite",
    "value" : "test_footer"
  }, {
    "name" : "subSuite",
    "value" : "TestFooter"
  }, {
    "name" : "host",
    "value" : "fv-az622-781"
  }, {
    "name" : "thread",
    "value" : "3345-MainThread"
  }, {
    "name" : "framework",
    "value" : "pytest"
  }, {
    "name" : "language",
    "value" : "cpython3"
  }, {
    "name" : "package",
    "value" : "tests.test_footer"
  }, {
    "name" : "resultFormat",
    "value" : "allure2"
  } ],
  "parameters" : [ {
    "name" : "URL",
    "value" : "'https://magento.softwaretestingboard.com/what-is-new.html'"
  } ],
  "links" : [ ],
  "hidden" : false,
  "retry" : false,
  "extra" : {
    "severity" : "normal",
    "retries" : [ ],
    "categories" : [ {
      "name" : "Test defects",
      "matchedStatuses" : [ ],
      "flaky" : false
    } ],
    "tags" : [ ]
  },
  "source" : "482a294cccb57025.json",
  "parameterValues" : [ "'https://magento.softwaretestingboard.com/what-is-new.html'" ]
}